### PR TITLE
Syncing dev with main

### PR DESCRIPTION
**Pull request by sestephens73**
Merging Will's changes back into dev so it's up to date

### CODE REVIEW

**Review by dialecticDolt — APPROVED** [review]
Thanks!

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "examples/nbody"]
 	path = examples/nbody
-	url = git@github.com:ut-parla/barnes-hut.git
+	url = https://github.com/ut-parla/barnes-hut.git
 [submodule "examples/blr"]
 	path = examples/blr
-	url = git@github.com:ut-parla/BLR.git
```
Comment (nicelhc13) [inline .gitmodules:6]: Just curious. Why is this change required?
Technically, both look fine and private repos enforces to use git@github.com format.
